### PR TITLE
Component status: new page & Quality Checklist on a per component basis

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Component status: new page & Quality Checklist on a per component  based

### CODE REVIEW

File: docs/components/QualityChecklist.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+export default function QualityChecklist({ component }: Props): Node {
+  const componentData = [
+    ...COMPONENT_DATA.buildingBlockComponents,
+    ...COMPONENT_DATA.generalComponents,
+    ...COMPONENT_DATA.utilityComponents,
+  ].find((cmpName) => cmpName.name === component);
+
+  return (
+    <MainSection name="Component Quality Checklist">
```
Comment (dangerismycat) [inline docs/components/QualityChecklist.js:21]: I think this should be "Component quality checklist"? Isn't that our preferred style?

**Review by dangerismycat — APPROVED** [review]

File: docs/components/PageHeaderQualitySummary.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+|};
+
+export default function PageHeaderQualitySumary({ name }: Props): Node {
+  const componentStatusData = [
+    ...COMPONENT_DATA.buildingBlockComponents,
+    ...COMPONENT_DATA.generalComponents,
+    ...COMPONENT_DATA.utilityComponents,
+  ].find((cmpName) => cmpName.name === name);
+
+  return componentStatusData?.status?.deprecated ? (
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/PageHeaderQualitySummary.js:19]: DONE

File: docs/components/StatusData.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { type Node } from 'react';
+import { Flex, Status, Text, Icon, Link } from 'gestalt';
+import { STATUS_EQUIVALENCY_MAP, STATUS_DESCRIPTION } from './COMPONENT_STATUS_MESSAGING.js';
+
+export default function StatusData({
+  type,
+  text,
+  href,
+}: {|
+  type: 'ready' | 'notAvailable' | 'partial' | 'planned' | 'deprecated',
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/StatusData.js:11]: DONE

File: docs/components/StatusData.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  text?: string,
+  href?: string,
+|}): Node {
+  let label = STATUS_DESCRIPTION[type].title;
+  if (text) {
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/StatusData.js:16]: DONE

File: docs/components/StatusData.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      )}
+      {href ? (
+        <Text size="200">
+          <Link underline="always" href={href}>
+            {label === 'Not available' ? 'N/A' : label}{' '}
+          </Link>
+        </Text>
+      ) : null}
+    </Flex>
+  ) : null;
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/StatusData.js:39]: DONE

File: docs/components/AccessibilityChecklist.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  return (
+    <Module.Expandable
+      accessibilityExpandLabel="Expand the module"
+      accessibilityCollapseLabel="Collapse the module"
+      id="accessibility-module"
+      items={[
+        {
+          children:
+            componentData?.status?.accessible?.summary === 'notAvailable' ? (
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/AccessibilityChecklist.js:37]: DONE

File: docs/components/AccessibilityChecklist.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+              <Table accessibilityLabel="Component Quality Checklist">
+                <colgroup>
+                  <col style={{ width: '20%' }} />
+                  <col style={{ width: '20%' }} />
+                  <col style={{ width: '60%' }} />
+                </colgroup>
+                <Box display="visuallyHidden">
+                  <Table.Header>
+                    <Table.Row>
+                      {['Quality item', 'Status', 'Status descripyion'].map((header) => (
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/AccessibilityChecklist.js:52]: DONE

File: docs/components/COMPONENT_STATUS_MESSAGING.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    title: 'iOS',
+    ready: 'Component is available in code for iOS.',
+    partial: '',
+    notAvailable: 'Component is not currently available in code for iOS.',
+    planned: 'Component is slotted to be built for iOS.',
+  },
+  responsive: {
+    shortTitle: 'Web',
+    title: 'Responsive Web',
+    ready: 'Component is available in code for web and mWeb.',
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/COMPONENT_STATUS_MESSAGING.js:53]: DONE

File: docs/components/QualityChecklist.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/components/QualityChecklist.js:21]: DONE

File: docs/components/AccessibilityChecklist.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  ].find((cmpName) => cmpName.name === component);
+
+  return (
+    <Module.Expandable
+      accessibilityExpandLabel="Expand the module"
+      accessibilityCollapseLabel="Collapse the module"
+      id="accessibility-module"
+      items={[
+        {
+          children:
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/AccessibilityChecklist.js:36]: DONE

**Pull request merged**